service/history/execution/mutable_state_builder.go

```diff
@@ -2659,6 +2659,28 @@ func (e *mutableStateBuilder) AddFailWorkflowEvent(
 	return event, nil
 }
 
+func (e *mutableStateBuilder) AddCancelWorkflowEvent(
```

Comment:
Is this different to `AddWorkflowExecutionCanceledEvent` on line 2757?

Reply:
yeah, we have these methods paired between historybuilder and mutablestatebuilder.  not entirely sure why, I just followed the patterns